File: c++/diff_1_4/hangingout.cpp
#include<iostream>

int main()
{
    int fireShift;
    int cases;

    int terranceCount = 0;
    int rejectCount = 0;

    std::cin >> fireShift;
    std::cin >> cases;
    
    for (int i = 0; i <= cases; i++)
    {
	std::string entExt;
	int amt;

	std::cin >> entExt;
	std::cin >> amt;

	if (entExt == "enter")
	{
	    if ((amt + terranceCount) > fireShift)
	    {
		rejectCount++;
	    }
	    else
	    {
		terranceCount += amt;
	    }
	}
	else
	{
	    terranceCount -= amt;
	}

    }

    std::cout << rejectCount;

    return 0;


}

File: python/diff_1_4/ hangingout.py

firesft, cases = input().split()


terrance_count = 0
reject_count = 0
for _ in range(int(cases)):
    ent_ex, amt = input().split()
    amt = int(amt)

    if ent_ex == "enter":
        if (amt + terrance_count) > int(firesft):
            reject_count +=1
        else:
            terrance_count += amt

    else:
        terrance_count -= amt
print(reject_count)

